### PR TITLE
Initialize from steady state

### PR DESCRIPTION
**Pull request by JonathanPham**
Addresses Issue #18.

This update enables the user to use the steady-state solution from an equivalent 0d model with steady BCs as the initial conditions for the original 0d model with pulsatile BCs. The user can use this setting for their 0d simulation by invoking the `-sic` or `--useSteadyIC` command line argument.

Note that this method works only for the following built-in BCs:
- flow
- pressure
- resistance
- RCR
- coronary

This method does not work with custom, user-defined BCs.

### CODE REVIEW

File: run_0d_solver_fast.py [hunk 1]
```diff
@@ -1114,19 +1122,52 @@ def set_up_and_run_0d_simulation(zero_d_solver_input_file_path, draw_directed_gr
...
     parameters = oneD_to_zeroD_convertor.extract_info_from_solver_input_file(zero_d_solver_input_file_path)
     parameters["check_jacobian"] = check_jacobian
+
+    if use_steady_soltns_as_ics:
+        parameters_mean = copy.deepcopy(parameters)
+        parameters_mean = use_mean_bcs.use_mean_values_for_bcs(parameters_mean)
+
+        # to run the 0d model with steady BCs to steady-state, simulate this model with large time step size for an arbitrarily large number of cardiac cycles
+        parameters_mean["number_of_time_pts_per_cardiac_cycle"] = 11
+        parameters_mean["number_of_cardiac_cycles"] = 100
```
Comment (mrp089) [inline run_0d_solver_fast.py]: Does it look like the steady-state solution is run for 100 cycles with 11 time steps each? Ideally, the solution should converge within a couple of time steps (when all C -> infinity and all L -> 0).

**Review by mrp089 — CHANGES_REQUESTED** [review]
I think the boundary conditions still need to be modified for the steady-state solution (i.e. convert to resistance only).

**Comment by JonathanPham** [comment]
Capacitance in RCR and Coronary BCs have been set to zero (represents "fully charged" capacitor) for steady simulation.

Also, `parameters_mean["number_of_cardiac_cycles"]` has been set to 3 for steady simulation (resistor-only simulation should converge within 2 cycles at most, but using 3 just in case).

Also, coronary BCs have been updated (all governing equations for the `OpenLoopCoronaryWithDistalPressureBlock` were divided by the intramyocardial capacitance (plus an epsilon) to prevent `NaN` solutions when capacitance is set to zero in steady simulation.

**Comment by JonathanPham** [comment]
Previously, we set the capacitance in the RCR and Coronary BCs to zero for steady simulation. In this process, we had to divide by the intramyocardial capacitance (plus an epsilon) in all governing equations for the `OpenLoopCoronaryWithDistalPressureBlock` to prevent `NaN` solutions when the intramyocardial capacitance is set to zero in steady simulation. However, we are concerned that this division may lead to poorly-behaved results. As such, we restored `OpenLoopCoronaryWithDistalPressureBlock` to its original formulation, where the intramyocardial capacitance is in the numerator, rather than the denominator, of the governing equations.

As such, in order to overcome issues with the `NaN` solutions in the coronary BCs when the intramyocardial capacitance is set to zero, we instead converted all RCR and Coronary BCs into equivalent resistance BCs for steady simulation (RCR and coronary BCs with zero capacitance are equivalent to pure resistance BCs).

However, when converting the RCR and coronary BCs into equivalent resistance BCs for steady simulation, we lost the internal variable originally associated with the RCR and coronary BCs. In the RCR BCs, this internal variable was the pressure proximal to the capacitor; in the coronary BCs, this internal variable was the volume of the intramyocardial capacitor. As such, in order to use the steady-state solutions from the steady simulation as the initial conditions for the original pulsatile model, we needed to "restore" the internal variables and set their values to be the steady-state values. This is done in the function, `use_steady_bcs.restore_internal_variables_for_capacitance_based_bcs()`.

A summary of the major changes is listed below:
1. added a default cardiac period of 1.0 seconds (to be used for 0d models where there is no explicit cardiac period set)
2. converted all RCR and Coronary BCs into equivalent resistance BCs for steady simulation
3. added function to 'restore' internal variables in RCR and coronary BCs for use in initialization

**Pull request merged**